src/systems/userinput/userinput.js

```diff
@@ -412,13 +412,20 @@ AFRAME.registerSystem("userinput", {
       gamepad && gamepadConnected({ gamepad });
     }
 
-    const retrieveXRGamepads = ({ session }) => {
-      for (const inputSource of session.inputSources) {
-        inputSource.gamepad.isWebXRGamepad = true;
-        inputSource.gamepad.targetRaySpace = inputSource.targetRaySpace;
-        inputSource.gamepad.primaryProfile = inputSource.profiles[0];
-        inputSource.gamepad.hand = inputSource.handedness;
-        gamepadConnected(inputSource);
+    const retrieveXRGamepads = ({ added, removed }) => {
```

Comment:
This line in `enter-vr` event listener calls `retrieveXRGamepads` with `session`.

https://github.com/mozilla/hubs/blob/fd81dd93f8b72c01d3d66e9e9458ee38b99386b5/src/systems/userinput/userinput.js#L447

Is it intentional? `added` and `removed` will be `undefined` so the function does nothing. Can't we simply drop the line?

Reply:
This ends up being a noop anyway since `inputsourceschange` will fire with the controllers anyway, but it does explain why I was seeing it called with just a session sometimes (which is why I added the `if(added)` and `if(removed)` checks. I think I can get rid of this and the checks.